File: AlgorithmNote/AlgorithmNote-Code/第3章 入门模拟/查找元素/B1004.cpp
#include <bits/stdc++.h>
using namespace std;
typedef struct {
    string name;
    string id;
    int score;
}student;
bool cmp(student s1, student s2) {//降序排列
    if(s1.score > s2.score) return true;
    else return false;
}
int main() {
    int n;
    cin >>n;
    student res[n];
    for(int i = 0; i < n; i++) {
        cin >>res[i].name >>res[i].id >> res[i].score;
    }
    sort(res, res + n, cmp);
    cout <<res[0].name <<" " <<res[0].id <<endl;
    cout <<res[n-1].name <<" " <<res[n-1].id <<endl;

    return 0;
}

File: AlgorithmNote/AlgorithmNote-Code/第3章 入门模拟/查找元素/B1004.py
mp = {}#dic初始化
n = eval(input())
for i in range(n):
    name, id, score = input().split()#分割出来的均是字符串
    name_id = f"{name} {id}"#格式拼接
    score = int(score)
    mp[score] = name_id
lmp = list(mp.keys())#dic的keys转换为list进行排序
lmp.sort()#升序排列
print(mp[lmp[n-1]])
print(mp[lmp[0]])